modules/core/src/main/java/org/locationtech/jts/coverage/CoverageRingEdges.java

```diff
@@ -216,6 +217,22 @@ private Set<Coordinate> findNodes(Geometry[] coverage) {
     return nodes;
   }
 
+  private Set<Coordinate> findBoundaryNodes(Set<LineSegment> lineSegments) {
+    Map<Coordinate, Integer> boundaryVertexCounter = new HashMap<>();
+    Set<Coordinate> nodes = new HashSet<Coordinate>();
+    for (LineSegment line : lineSegments) {
+      int count = boundaryVertexCounter.getOrDefault(line.p0, 0);
+      boundaryVertexCounter.put(line.p0, count+1);
+      count = boundaryVertexCounter.getOrDefault(line.p1, 0);
```

Comment:
Definining a new local var would make this logic clearer

Reply:
Do you mean a new local variable for the second count ? I can do it for sure. Not sure it would make the logic clearer though as the variable count all points with the same coordinate, either p0 or p1.
Indeed, I try to follow the JTS style which is mostly based on java5 I think ,but from my point of view, a more readable version would be : 
  private Set<Coordinate> findBoundaryNodes(Set<LineSegment> lineSegments) {
    Map<Coordinate, Integer> counter = new HashMap<>();
    for (LineSegment line : lineSegments) {
      counter.put(line.p0, counter.getOrDefault(line.p0, 0) + 1);
      counter.put(line.p1, counter.getOrDefault(line.p1, 0) + 1);
    }
    return counter.entrySet().stream()
        .filter(e->e.getValue()>2)
        .map(Map.Entry::getKey).collect(Collectors.toSet());
  }

Reply:
That looks even better!